build/main_test.go

```diff
@@ -34,3 +34,45 @@ func ExampleTransactionBuilder() {
 	fmt.Printf("tx base64: %s", txeB64)
 	// Output: tx base64: AAAAADZY/nWY0gx6beMpf4S8Ur0qHsjA8fbFtBzBx1cbQzHwAAAAZAAAAAAAAAABAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAALSRpLtCLv2eboZlEiHDSGR6Hb+zZL92fbSdNpObeE0EAAAAAAAAAAB3NZQAAAAAAAAAAARtDMfAAAABA2oIeQxoJl53RMRWFeLB865zcky39f2gf2PmUubCuJYccEePRSrTC8QQrMOgGwD8a6oe8dgltvezdDsmmXBPyBw==
 }
+
+// ExamplePathPayment creates and signs a simple transaction with PathPayment operation, and then
+// encodes it into a base64 string capable of being submitted to stellar-core.
+func ExamplePathPayment() {
```

Comment:
My more drastic proposal to resolve the source/dest ambiguity:

```go
SendNative{Amount: "100.0"}
SendCredit{Code:"USD", Issuer:"BLAH" Amount:"100.0"}
PayWithNative{MaxAmount: "100.0"}
PayWithCredit{Code:"USD", Issuer:"BLAH", MaxAmount: "100.0"}
```

And then one additional proposal about how we might expose pathing:

```
// manually specified path
Payment(
  Destination{addy},
  PayWithNative{MaxAmount: "100.0"}.
    ThroughCredit("EUR", "ISSUER").
    ThroughCredit("USD", "SOMEONEELSE"),  
)

// Path finding
Payment(
  horizon.FindPath(
    SendNative{Amount: "100.0"}
    PayWithCredit{Code:"USD", Issuer:"BLAH"} // MaxAmount will be filled automatically if a path is found using USD this source asset
  ),
)
```

Thoughts?